### PR TITLE
AQCU-1228/1223: changes to explanation on DC

### PR DESCRIPTION
**Pull request by mhines-usgs**
AQCU-1228:  Rename:
"Processor Types" --> "Dataset Processor Types"
"Primary Timeseries" --> "Selected Time Series"
"Upchain/downchain Timeseries" --> "Upchain/Downchain Time Series"
Add under the "Explanation" title add a note that "TS = Time Series" since "TS" is used in the explanation.

AQCU-1223: Dynamic dashed line

### CODE REVIEW

File: inst/templates/derivationchain/custom.js [hunk 1]
```diff
@@ -368,8 +368,15 @@ function generateLegend(nodes) {
   var visibleProcessors = [];
   var addExternalLine = false;
   var legendContainer = $("#legendContainer");
-  $(legendContainer).html('<strong>Explanation</strong><br/><br/><span style="text-decoration:underline">Processor Types:</span> <br/>');
-
+  $(legendContainer).html('<strong>Explanation</strong><br/><br/><span style="text-decoration:underline">Dataset Processor Types:</span> <br/>');
+  
+  var flags = [], output = [], l = nodes.length, i;
```
Comment (mhines-usgs) [inline inst/templates/derivationchain/custom.js]: this loops through the node locations and if there are different locations in the list, add to another array which we'll check later before adding that dashed line to the legend

File: inst/templates/derivationchain/custom.js [hunk 2]
```diff
@@ -368,8 +368,15 @@ function generateLegend(nodes) {
...
+    for( i=0; i<l; i++) {
+      if( flags[nodes[i].data["location"]]) continue;
+      flags[nodes[i].data["location"]] = true;
+      output.push(nodes[i].data["location"]);
+    }
```
Comment (zmoore-usgs) [inline inst/templates/derivationchain/custom.js]: I might be incorrectly understanding what this is doing but I think this process could mostly happen within the loop below this one?

If you create an array of added locations before the loop and add each node's location to that array if it's not already in the array and then after the loop you can have if(locations.length>1){addExternalLine = true;}

That way we won't be running through a loop of all of the nodes twice. The overall approach of your ticket seems good though!

Comment (mhines-usgs) [reply inst/templates/derivationchain/custom.js]: Ok, I'll try that, sorry to flood you with PRs!

**Comment by mhines-usgs** [comment]
Seems to work fine, thanks for the suggestion @zmoore-usgs !

**Pull request merged**